kafka-admin/src/main/java/org/bf2/admin/kafka/admin/RecordOperations.java

```diff
@@ -137,6 +138,9 @@ public ConsumerRecords<byte[], byte[]> next() {
                     int pollSize = records.count();
                     emptyPoll = pollSize == 0;
                     recordsConsumed.addAndGet(pollSize);
+                    if (recordsConsumed == null) {
+                        throw new NoSuchElementException("No records were consumed");
+                    }
```

Comment:
`recordsConsumed` will never be null (assigned once on line 125). I think the simplest thing for this exception would be to put the following as the first thing in this `next` method:

```java
if (!hasNext()) {
    throw new NoSuchElementException();
}
```

Reply:
Okaay! Makes sense. When I tried with `if ( emptyPoll ) { throw new NoSuchElementException(); }`  it was throwing an incorrect status code for the other system tests. So, I tried this one.  But it makes sense to have the `!hasNext` condition for the exception. 
Thanks Mike!

Reply:
Sounds good, thanks. If the systemtests have issues with the change, let me know. They might be incorrect.

Reply:
That worked perfectly Mike! Thanks again.